starter/share/modules1/detonators_mod.F

```diff
@@ -80,59 +80,136 @@ MODULE DETONATORS_MOD
+          my_real :: ZDET
+          my_real :: NX
+          my_real :: NY
+          my_real :: NZ
+        END TYPE DETONATOR_PLANE_STRUCT_
+
+        ! --- /DFS/WAVE_SHAPER DATA STRUCTURE
+        TYPE DETONATOR_WAVE_SHAPER_STRUCT_
+          LOGICAL :: IS_MAT_VALID = .FALSE.
+          INTEGER :: MAT = 0
+          INTEGER :: NUMNOD
+          my_real :: XDET
+          my_real :: YDET
+          my_real :: ZDET
+          my_real :: VDET
+          my_real :: TDET
+          INTEGER,DIMENSION(:),ALLOCATABLE :: NODES
+          my_real,DIMENSION(:),ALLOCATABLE :: TIME
+          INTEGER,DIMENSION(:),ALLOCATABLE :: FLAG
+          INTEGER,DIMENSION(:),ALLOCATABLE :: IORDR
+        END TYPE DETONATOR_WAVE_SHAPER_STRUCT_
+
+        ! --- /DFS/DETCORD DATA STRUCTURE
+        TYPE DETONATOR_CORD_STRUCT_
+          LOGICAL :: IS_MAT_VALID = .FALSE.
+          INTEGER :: MAT = 0
+          INTEGER :: IOPT
+          INTEGER :: NUMNOD
+          INTEGER,DIMENSION(:),ALLOCATABLE :: NODES
+          my_real,DIMENSION(:),ALLOCATABLE :: TDET_PATH
+          my_real :: TDET
+          my_real :: VDET
+        END TYPE DETONATOR_CORD_STRUCT_
+
+        ! --- MAIN DATA STRUCTURE
+        TYPE DETONATORS_STRUCT_
+          !numbering
+          INTEGER :: N_DET !sum of all detonators
+          INTEGER :: N_DET_POINT !number of /DFS/DETPOINT
+          INTEGER :: N_DET_LINE !number of /DFS/DETLINE
+          INTEGER :: N_DET_WAVE_SHAPER !number of /DFS/WAVE_SHAPER
+          INTEGER :: N_DET_CORD !number of /DFS/DETCORD
+          INTEGER :: N_DET_PLANE !number of /DFS/DETPLANE
+          !all detonators
+          TYPE(DETONATOR_POINT_STRUCT_), DIMENSION(:), ALLOCATABLE :: POINT
+          TYPE(DETONATOR_LINE_STRUCT_), DIMENSION(:), ALLOCATABLE :: LINE
+          TYPE(DETONATOR_CORD_STRUCT_), DIMENSION(:), ALLOCATABLE :: CORD
+          TYPE(DETONATOR_WAVE_SHAPER_STRUCT_), DIMENSION(:), ALLOCATABLE :: WAVE_SHAPER
+          TYPE(DETONATOR_PLANE_STRUCT_), DIMENSION(:), ALLOCATABLE :: PLANE
+        END TYPE DETONATORS_STRUCT_
         
       CONTAINS
-      
-      
-C-----------------------------------------------   
+
 Chd|====================================================================
 Chd|  DEALLOCATE_DETONATORS         share/modules1/detonators_mod.F
 Chd|-- called by -----------
 Chd|        LECTUR                        source/starter/lectur.F       
 Chd|-- calls ---------------
 Chd|====================================================================
-      SUBROUTINE DEALLOCATE_DETONATORS(DET)
+      SUBROUTINE DEALLOCATE_DETONATORS(DETONATORS)
 C-----------------------------------------------
 C   I m p l i c i t   T y p e s
 C-----------------------------------------------
 #include      "implicit_f.inc"
 C-----------------------------------------------
 C   D u m m y   A r g u m e n t s
 C-----------------------------------------------
-      TYPE(DETONATOR_STRUCT_)::DET      
+      TYPE(DETONATORS_STRUCT_)::DETONATORS
```

Comment:
INTENT statement is missing

Reply:
The missing INTENT statement was added in detonators_mod.F
I also added de INTENT statement in reader subroutines. The argument has already existed without INTENT statement (it was renamed in this commit).